compiler/src/iree/compiler/GlobalOptimization/RaiseSpecialOps.cpp

```diff
@@ -679,6 +679,75 @@ static Value rewriteCatNegateAndSlice(RewriterBase &rewriter,
   return createCatNegateAndSlice(rewriter, outTensor, source);
 }
 
+// Match an extract slice that has a shape of size N and an output shape of size
+// N-1, where the extraction happened along only the innermost dimension.
+//
+// e.g.
+// %extracted_slice = tensor.extract_slice
+//   %1[0, 0, 0, 1] [1024, 7, 7, 1] [1, 1, 1, 1]
+//   : tensor<1024x7x7x2xf32> to tensor<1024x7x7xf32>
+// Rewrite by adding a transpose that makes dim N the outermost and slice along
+// that dim
+static void matchAndRewriteInnerExtractSlice(RewriterBase &rewriter,
+                                             tensor::ExtractSliceOp sliceOp) {
```

Comment:
Can we move this pattern here: https://github.com/iree-org/iree/blob/58feff319e2fd0dff7909358741a26ffa5807823/compiler/src/iree/compiler/GlobalOptimization/DecomposeConcat.cpp#L96

And keep it under the same flag? Maybe rename the `DecomposeConcat` pass also for clarity.